connectivity/FEATURE_BLE/source/generic/SecurityManagerImpl.cpp

```diff
@@ -577,14 +794,19 @@ ble_error_t SecurityManager::getLinkEncryption(
     link_encryption_t *encryption
 )
 {
-    if (!_db) return BLE_ERROR_INITIALIZATION_INCOMPLETE;
+    if (!_db) {
+        tr_error("Failed to retrieve link encryption, DB not initialized");
+        return BLE_ERROR_INITIALIZATION_INCOMPLETE;
+    }
     ControlBlock_t *cb = get_control_block(connection);
     if (!cb) {
+        tr_error("Failed to retrieve link encryption, no CB for %d", connection);
         return BLE_ERROR_INVALID_PARAM;
     }
 
     SecurityDistributionFlags_t* flags = _db->get_distribution_flags(cb->db_entry);
     if (!flags) {
+        tr_error("Failed to retrieve link encryption, empty distribution flags for %d", connection);
```

Comment:
It is present in many places, it can happen if the database is corrupted. I wonder if we shouldn't check it when it is loaded then assume it is consistent.